Source/WebCore/page/PointerLockController.h

```diff
@@ -71,7 +73,9 @@ class PointerLockController {
     bool m_lockPending { false };
     bool m_unlockPending { false };
     bool m_forceCursorVisibleUponUnlock { false };
+    bool m_unadjustedMovement { false };
```

Comment:
Why not store a `std::optional<PointerLockOptions> m_pointerLockOptions;` instead?

Reply:
It's essentially the same as storing a boolean flag but I think it conveys the meaning better and we get any future options for free.

Reply:
Sure, will update in next version of the patch.

Reply:
The problem I'm running into with doing this is I can't have a `std::optional<PointerLockOptions>` member in PointerLockController without doing `#include "PointerLockOptions.h"` in PointerLockController.h, and PointerLockController.h is a private header in WebCore, but PointerLockOptions.h is an internal header. So code outside of WebCore like WebChromeClient.mm can't find PointerLockOptions.h.

```
In file included from /Users/jahoward/git/ext/WebKit/Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:112:
/Users/jahoward/git/ext/WebKit/WebKitBuild/Debug/WebCore.framework/PrivateHeaders/PointerLockController.h:29:10: fatal error: 'PointerLockOptions.h' file not found
#include "PointerLockOptions.h"
```

So the options (heh) seem to be:

a. Make PointerLockOptions.h a private header in WebCore.framework
b. Stay with the boolean

I kinda prefer b. What do you think?

Reply:
I prefer (a) mainly because it makes plumbing simpler - we just pass the options object (as a side effect it resembles the `requestPointerLock` web API more closely...). (a) shouldn't be too disruptive a change, right?

Reply:
OK sure, I will implement (a) and update the patch. I just wasn't sure if adding a new private header to WebCore was OK here, but sounds like it's no problem.

Reply:
OK, this is done in the latest patch.